psi4/driver/driver_cbs.py

```diff
@@ -985,65 +1385,173 @@ def _validate_cbs_inputs(cbs_metadata, molecule):
         # 1a) all items must have wfn
         if "wfn" not in item:
             raise ValidationError(f"Stage {iitem} doesn't have defined level of theory!")
-    # 1b) all items must have basis set
+        # 1b) all items must have basis set
         if "basis" not in item:
             raise ValidationError(f"Stage {iitem} doesn't have defined basis sets!")
-    # 2a) process required stage parameters and assign defaults
+        # 2) process required stage parameters and assign defaults
         stage = {}
         stage["wfn"] = item["wfn"].lower()
         stage["basis"] = _expand_bracketed_basis(item["basis"].lower(), molecule)
-        # 2b) if first item is not HF, generate it
-        if len(metadata) == 0 and stage["wfn"] not in ["hf", "c4-hf", "scf", "c4-scf"]:
-            scf = {}
-            if stage["wfn"].startswith("c4"):
-                scf["wfn"] = "c4-hf"
-            else:
-                scf["wfn"] = "hf"
-            scf["basis"] = ([stage["basis"][0][-1]], [stage["basis"][1][-1]])
-            scf["treatment"] = "scf"
-            scf["stage"] = "scf"
-            scf["scheme"] = _get_default_xtpl(len(scf["basis"][1]), scf["treatment"])
-            scf["alpha"] = None
-            scf["options"] = False
-            scf["options_lo"] = False
-            metadata.append(scf)
-    # 2c) keep processing current stage
-        stage["treatment"] = item.get("treatment", "scf" if len(metadata) == 0 else "corl")
-        stage["stage"] = item.get("stage", False)
-        if not stage["stage"]:
+        # 2a) process DFT methods
+        if stage["wfn"] in functionals and stage["wfn"] not in ["hf", "scf"]:
+            # 2ai) first stage - split into components, unless "component" == "dft"
+            #     This allows users to request "BLYP-D3(0)/pc-[12]" without 
+            # specifying extra parameters. If "component" is not "dft", the "fctl", 
+            # "disp", "dh", and "nl" stages are generated automatically as required, 
+            # and the "DFT TOTAL ENERGY" is not used.
+            #     In cases where "component" is "dft", that setting gets processed,
+            # and further generation of component stages is skipped.
+            #     Note that none of these stages are delta-stages, as only one E/G/H
+            # call is made and the components are picked from the returned wfn
+            # automatically.
             if len(metadata) == 0:
-                stage["stage"] = "scf"
-            elif len(metadata) == 1:
-                stage["stage"] = "corl"
+                if "component" not in item:
+                    fctl = {
+                        "wfn": stage["wfn"],
+                        "basis": stage["basis"],
+                        "component": f'{stage["wfn"]}-fctl',
```

Comment:
Would a user possibly want to set this sort of information manually? What if they want a different scheme for "nl" as for "disp"?

Reply:
Yes, that ought to be possible. Note that this big `if`-block only deals with the first stage.